supported/osg-htc/osg-hosted-ce/templates/configmap.yaml

```
@@ -77,7 +77,8 @@ data:
     # Templatize IDTOKEN generation for glidein -> CE collector advertising (SOFTWARE-5556)
     {{- range $index, $map := .Values.SciTokenRemoteUserMapping }}
     {{- range $url, $user := $map }}
-    JOB_ROUTER_CREATE_IDTOKEN_{{ $user }} @=end
+    # N.B. Special chars in a helm template regex need to be escaped with '\\'
+    JOB_ROUTER_CREATE_IDTOKEN_{{ mustRegexReplaceAll "[\\.-]" "$user" "_" }} @=end
```

```suggestion
    JOB_ROUTER_CREATE_IDTOKEN_{{ mustRegexReplaceAll "[^A-Za-z0-9_]" "$user" "_" }} @=end
```
Why not replace all invalid chars instead of playing whack-a-mole with special characters as we find them?

---

Sure. I just figured we only had to deal with the legal Unix char set